test/core/expectations/compound_data/test_expectations_cursor.pkb

```diff
@@ -1540,17 +1551,150 @@ Diff:%
       from dual connect by level <=2 order by rownum desc;
     
     --Act
-    ut3.ut.expect(l_actual).to_equal(l_expected).join_by('COLVAL/IDS');
-    --Assert
-    --Assert
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by('COLVAL/IDS'); 
+    
  l_expected_message := q'[%Actual: refcursor [ count = 2 ] was expected to equal: refcursor [ count = 2 ]%
 Diff:%
 %Unable to join sets:%
-%Join key /*/COLVAL/IDS does not exists in expected%
-%Join key /*/COLVAL/IDS does not exists in actual%]';
+%Join key COLVAL/IDS does not exists in expected%
+%Join key COLVAL/IDS does not exists in actual%]';
     l_actual_message := ut3.ut_expectation_processor.get_failed_expectations()(1).message;
     --Assert
     ut.expect(l_actual_message).to_be_like(l_expected_message);
   end;
+ 
+  procedure compare_nest_tab_col_jb is
+    l_actual   sys_refcursor;
+    l_expected sys_refcursor;
+    l_actual_tab ut3.ut_key_value_pairs := ut3.ut_key_value_pairs();
+    l_expected_tab ut3.ut_key_value_pairs := ut3.ut_key_value_pairs();
+    l_expected_message varchar2(32767);
+    l_actual_message   varchar2(32767);
+  begin
+    select ut3.ut_key_value_pair(rownum,'Something '||rownum)
+    bulk collect into l_actual_tab
+    from dual connect by level <=2;
+ 
+    select ut3.ut_key_value_pair(rownum,'Something '||rownum)
+    bulk collect into l_expected_tab
+    from dual connect by level <=2;
+      
+    --Arrange
+    open l_actual for select key,value
+      from table(l_actual_tab) order by 1 asc;
+
+    open l_expected for select key,value
+      from table(l_expected_tab) order by 1 desc;
+    
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by('KEY');
```

Comment:
What will happen when your cursor `l_actual` is:
```sql
select rownum rn, l_actual_tab
  from dual connect by level <=2;
ut3.ut.expect(l_actual).to_equal(l_expected).join_by('RN,L_ACTUAL_TAB/KEY');
```
The dataset is then something like:
```xml
<?xml version="1.0"?>
<ROWSET>
 <ROW>
  <RN>1</RN>
  <L_ACTUAL_TAB>
   <UT_KEY_VALUE_PAIR>
    <KEY>1</KEY>
    <VALUE>Something 1</VALUE>
   </UT_KEY_VALUE_PAIR>
   <UT_KEY_VALUE_PAIR>
    <KEY>2</KEY>
    <VALUE>Something 2</VALUE>
   </UT_KEY_VALUE_PAIR>
  </L_ACTUAL_TAB>
 </ROW>
 <ROW>
  <RN>2</RN>
  <L_ACTUAL_TAB>
   <UT_KEY_VALUE_PAIR>
    <KEY>1</KEY>
    <VALUE>Something 1</VALUE>
   </UT_KEY_VALUE_PAIR>
   <UT_KEY_VALUE_PAIR>
    <KEY>2</KEY>
    <VALUE>Something 2</VALUE>
   </UT_KEY_VALUE_PAIR>
  </L_ACTUAL_TAB>
 </ROW>
</ROWSET>
```
That will not be (and probably should not be) supporter but might be still worth documenting in test.
We just don't know what will happen as there is not test case to document how the `ut.expect` will behave.